docs/developer/community/contributing.mdx

```diff
@@ -37,6 +34,163 @@ For compatibility, Saleor also provides `requirements.txt` and `requirements_dev
+- etc.
+
+### API file structure
+The `API` files are in `saleor/graphql/` directory. Every app has its own directory 
+inside with:
+- `schema.py` - with definitions of queries and mutations
+- `sorters.py` - with definitions of sorters
+- `filters.py` - with definitions of filters
+- `types.py` - with definitions of corresponding types
+- `enums.py` - with related enums
+- `dataloaders.py` - with data loaders for the given module.
+- mutations file or directory.
+- test directory.
+
+We are aiming to have `mutations` directory in every module with a file per every mutation
+how it's done in the `checkout` directory. See an example below:
+
+```bash
+.
+└── saleor
+    └── graphql
+        └── checkout
+            ├── migrations
+            │   └── checkout_create.py
+            │   └── checkout_complete.py
+            └── schema.py
+```
+
+### Tests file structure
+We are keeping tests in `tests` directory. We are aiming to keep queries and mutations
+in separate directories with a single file for every query and mutation. 
+After joining it with the previous example, it would look like this:
+
+```bash
+.
+└── saleor
+    └── graphql
+        └── checkout
+            ├── migrations
+            │   └── checkout_create.py
+            │   └── checkout_complete.py
+            ├── schema.py
+            └── tests
+                ├── migrations
+                │   ├── test_checkout_complete.py
+                │   └── test_checkout_create.py
+                └── queries
+                    └── test_checkout.py
+```
+
+
+## Testing
+Testing are really important part of every project.
+In Saleor we're using the `pytest` library and we mostly have unit tests.
+
+The test file structure was introduced in the [tests file structure](contributing#tests-file-structure) part.
+The main rule is not overloading test files. Always smaller files are preferable over big ones
+where lots of logic is tested and it's hard to extend.
+In the case of testing `API` we would like to split all tests into `mutations` and `queries`
+section and test every query and mutation in a separate file.
+
+### How to run tests?
+To run tests just enter `pytest` in your terminal.
+```bash
+pytest
+```
+
+To speed time testing time, we recommend using the `reuse-db` flag.
+```bash
+pytest --reuse-db
+```
+
+### How to run particular tests?
+As running all tests is quite time-consuming, sometimes you would like to run
+only tests from one dictionary or even just a particular test.
+You can use for that the following commands. In the case of a particular directory or file, 
+just provide the path after the `pytest` command, like:
+```bash
+pytest --reuse-db saleor/graphql/app/tests
+```
+
+If you would like to run a particular test, you need to provide the path to the file
+where the test is, and the file name after the `::` sign. In the case of running a single test
+it's also worth using `-n0` flag to run the test only in one thread. It will significantly decrease time.
```

Comment:
Is it a typo or should it be `-n 0`? I always use `-n 0`.

Reply:
I'm using `-n0`, so I suppose there is no difference 😄